lib/api/api_auth.go

```diff
@@ -353,3 +368,124 @@ func iso88591ToUTF8(s []byte) []byte {
 	}
 	return []byte(string(runes))
 }
+
+type tokenManager struct {
```

Comment:
Optional/opinionated: I'd move the tokenManager code to it's own file - it's a non-trivial amount of code and used both here and in the csrf file/code.